### ISSUE TITLE
SWC cannot running with lodash and swagger

### ISSUE THREAD

**Issue by Kennytian** [nestjs/nest#11856]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

```

/Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:3032
      return object && baseFor(object, iteratee, keys);
                       ^
TypeError: Cannot destructure property 'prototype' of 'param.type' as it is undefined.
    at /Users/kenny/node_modules/.pnpm/@nestjs+swagger@7.0.2_@nestjs+common@10.0.2_@nestjs+core@10.0.2_class-transformer@0.5.1_class_it2dqwtbh5prjmfaj6khbziqr4/node_modules/@nestjs/swagger/dist/services/parameters-metadata-mapper.js:25:21
    at /Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:3585:27
    at /Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:4967:15
    at baseForOwn (/Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:3032:24)
    at /Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:4936:18
    at baseMap (/Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:3584:7)
    at map (/Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:9622:14)
    at flatMap (/Users/kenny/node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js:9325:26)
    at ParametersMetadataMapper.transformModelToProperties (/Users/kenny/node_modules/.pnpm/@nestjs+swagger@7.0.2_@nestjs+common@10.0.2_@nestjs+core@10.0.2_class-transformer@0.5.1_class_it2dqwtbh5prjmfaj6khbziqr4/node_modules/@nestjs/swagger/dist/services/parameters-metadata-mapper.js:13:49)
    at exploreApiParametersMetadata (/Users/kenny/node_modules/.pnpm/@nestjs+swagger@7.0.2_@nestjs+common@10.0.2_@nestjs+core@10.0.2_class-transformer@0.5.1_class_it2dqwtbh5prjmfaj6khbziqr4/node_modules/@nestjs/swagger/dist/explorers/api-parameters.explorer.js:25:70)
```

### Minimum reproduction code

https://github.com/Kennytian/nestjs10-swc-swagger-issue 

### Steps to reproduce

pnpm i --save-dev @swc/cli @swc/core

nest start

### Expected behavior

No error

### Package

- [ ] I don't know. Or some 3rd-party package
- [ ] <code>@nestjs/common</code>
- [ ] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [ ] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### NestJS version

10.0.2

### Packages versions

```json
 "dependencies": {
    "@nestjs/axios": "^3.0.0",
    "@nestjs/common": "^10.0.2",
    "@nestjs/config": "^3.0.0",
    "@nestjs/core": "^10.0.2",
    "@nestjs/platform-express": "^10.0.2",
    "@nestjs/swagger": "^7.0.2",
    "@nestjs/throttler": "^4.1.0",
    "axios": "^1.4.0",
    "bignumber.js": "^9.1.1",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.14.0",
    "cross-env": "^7.0.3",
    "crypto-js": "^4.1.1",
    "dayjs": "^1.11.8",
    "dotenv": "^16.3.1",
    "express": "^4.18.2",
    "helmet": "^7.0.0",
    "jsonwebtoken": "^9.0.0",
    "nanoid": "^3.3.6",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^5.0.1",
    "rxjs": "^7.8.1",
    "swagger-ui-express": "^4.6.3",
    "uuid": "^9.0.0"
  },
  "devDependencies": {
    "@commitlint/cli": "^17.6.5",
    "@commitlint/config-conventional": "^17.6.5",
    "@nestjs/cli": "^10.0.3",
    "@swc/cli": "^0.1.62",
    "@swc/core": "^1.3.66",
    "@types/express": "^4.17.17",
    "@types/jest": "^29.5.2",
    "@types/multer": "^1.4.7",
    "@types/node": "^20.3.1",
    "@typescript-eslint/eslint-plugin": "^5.59.11",
    "@typescript-eslint/parser": "^5.60.0",
    "eslint": "^8.43.0",
    "eslint-config-prettier": "^8.8.0",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-simple-import-sort": "^10.0.0",
    "eslint-plugin-unused-imports": "^2.0.0",
    "jest": "^29.5.0",
    "lint-staged": "^13.2.2",
    "prettier": "^2.8.8",
    "source-map-support": "^0.5.21",
    "ts-jest": "^29.1.0",
    "ts-loader": "^9.4.3",
    "ts-node": "^10.9.1",
    "tsconfig-paths": "^4.2.0",
    "typescript": "^5.1.3"
  },
```

### Node.js version

v18.16.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by micalevisk** [comment]
Please provide a minimum reproduction repository. You can start from [this minimal project](https://stackblitz.com/edit/nestjs-issue-minimum-repro-starter?devtoolsheight=33).

## [why reproductions are required](https://antfu.me/posts/why-reproductions-are-required)

**Comment by Kennytian** [comment]
> Please provide a minimum reproduction repository. You can start from [this minimal project](https://stackblitz.com/edit/nestjs-issue-minimum-repro-starter?devtoolsheight=33).
> 
> ## [why reproductions are required](https://antfu.me/posts/why-reproductions-are-required)

## This minimal project
https://github.com/Kennytian/nestjs10-swc-swagger-issue


If I add @Headers into method getHello, throw the below post error

```
TypeError: Cannot destructure property 'prototype' of 'param.type' as it is undefined

```

**Comment by kamilmysliwiec** [comment]
Fixed in `@nestjs/swagger@7.0.3`. Thanks for reporting!
